ecosystem/typescript/sdk/src/aptos_client.ts

```diff
@@ -389,61 +389,64 @@ export class AptosClient {
   }
 
   /**
-   * Waits up to 10 seconds for a transaction to move past pending state.
-   * @param txnHash A hash of transaction
-   * @returns A Promise, that will resolve if transaction is accepted to the
-   * blockchain and reject if more then 10 seconds passed
+   * Wait for a transaction to move past pending state.
+   *
+   * There are 3 possible outcomes of transaction processing as pertaining to
+   * this function:
+   * 1. Transaction is processed and successfully committed to the blockchain.
+   * 2. Transaction is processed but rejected for some reason, and is therefore
+   *    not committed to the blockchain.
+   * 3. Transaction is not processed within the specified timeout.
+   *
+   * In case 1, this function resolves with the transaction response returned
+   * by the API.
+   *
+   * In case 2, if `checkSuccess` if false (the default), this function returns
+   * the transaction response just like in case 1, in which the `success` field
+   * will be false. If `checkSuccess` is true, it will instead throw a
+   * TransactionNotCommittedError.
```

Comment:
We're making up new terms here.  @rajkaramchedu isn't this a `FailedTransaction`?

There are 5 actual outcomes of submitting transaction:
* Successful and committed
* Failed and committed
* Rejected and not committed
* Expired and not committed
* My polling loop for checking if the transaction is committed timed out, and it might be committed eventually

Reply:
Okay let me clear this up. I can also call it a `FailedTransactionError`, and make it clear that it comprises a few of these cases.

Reply:
I rewrote it a bit. I said there are 4 outcomes because (correct me if I'm wrong) "expired and not commited" is just a specific case of "rejected and not committed".